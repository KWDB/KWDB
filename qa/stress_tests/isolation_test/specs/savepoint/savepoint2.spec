setup
{
  CREATE TABLE foo (
     key INT PRIMARY KEY,
     value INT
  );

  INSERT INTO foo VALUES (1, 1);
}

teardown
{
  DROP TABLE foo;
}

session s1
step s1			{ BEGIN transaction isolation level read committed; }
step s1r1		{ SELECT * FROM foo FOR KEY SHARE; }
step s1s		{ SAVEPOINT f; }
step s1r2		{ SELECT * FROM foo FOR NO KEY UPDATE; }
step s1rs		{ ROLLBACK TO f; }
step s1c		{ COMMIT; }

session s2
step s2			{ BEGIN transaction isolation level read committed; }
step s2r1		{ SELECT * FROM foo FOR UPDATE; }
step s2r2		{ SELECT * FROM foo FOR NO KEY UPDATE; }
step s2c		{ COMMIT; }

permutation s1 s2 s1r1 s1s s1r2 s1rs s2r1 s1c s2c
permutation s1 s2 s1r1 s1s s1r2 s2r1 s1rs s1c s2c
permutation s1 s2 s1r1 s1s s1r2 s1rs s2r2 s1c s2c
permutation s1 s2 s1r1 s1s s1r2 s2r2 s1rs s1c s2c