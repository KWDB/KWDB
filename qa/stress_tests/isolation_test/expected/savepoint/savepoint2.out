Parsed test spec with 2 sessions

starting permutation: s1 s2 s1r1 s1s s1r2 s1rs s2r1 s1c s2c
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s1r1: SELECT * FROM foo FOR KEY SHARE;  finished.
key|value
---+-----
  1|    1
(1 row)

step s1s: SAVEPOINT f;  finished.
step s1r2: SELECT * FROM foo FOR NO KEY UPDATE;  finished.
key|value
---+-----
  1|    1
(1 row)

step s1rs: ROLLBACK TO f;  finished.
step s2r1: SELECT * FROM foo FOR UPDATE; <waiting ...>
step s1c: COMMIT;  finished.
step s2r1: <... completed>
key|value
---+-----
  1|    1
(1 row)

step s2c: COMMIT;  finished.

starting permutation: s1 s2 s1r1 s1s s1r2 s2r1 s1rs s1c s2c
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s1r1: SELECT * FROM foo FOR KEY SHARE;  finished.
key|value
---+-----
  1|    1
(1 row)

step s1s: SAVEPOINT f;  finished.
step s1r2: SELECT * FROM foo FOR NO KEY UPDATE;  finished.
key|value
---+-----
  1|    1
(1 row)

step s2r1: SELECT * FROM foo FOR UPDATE; <waiting ...>
step s1rs: ROLLBACK TO f;  finished.
step s1c: COMMIT;  finished.
step s2r1: <... completed>
key|value
---+-----
  1|    1
(1 row)

step s2c: COMMIT;  finished.

starting permutation: s1 s2 s1r1 s1s s1r2 s1rs s2r2 s1c s2c
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s1r1: SELECT * FROM foo FOR KEY SHARE;  finished.
key|value
---+-----
  1|    1
(1 row)

step s1s: SAVEPOINT f;  finished.
step s1r2: SELECT * FROM foo FOR NO KEY UPDATE;  finished.
key|value
---+-----
  1|    1
(1 row)

step s1rs: ROLLBACK TO f;  finished.
step s2r2: SELECT * FROM foo FOR NO KEY UPDATE; <waiting ...>
step s1c: COMMIT;  finished.
step s2r2: <... completed>
key|value
---+-----
  1|    1
(1 row)

step s2c: COMMIT;  finished.

starting permutation: s1 s2 s1r1 s1s s1r2 s2r2 s1rs s1c s2c
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s1r1: SELECT * FROM foo FOR KEY SHARE;  finished.
key|value
---+-----
  1|    1
(1 row)

step s1s: SAVEPOINT f;  finished.
step s1r2: SELECT * FROM foo FOR NO KEY UPDATE;  finished.
key|value
---+-----
  1|    1
(1 row)

step s2r2: SELECT * FROM foo FOR NO KEY UPDATE; <waiting ...>
step s1rs: ROLLBACK TO f;  finished.
step s1c: COMMIT;  finished.
step s2r2: <... completed>
key|value
---+-----
  1|    1
(1 row)

step s2c: COMMIT;  finished.
